api/src/events.rs

```diff
@@ -77,7 +77,8 @@ impl EventsApi {
         // Ensure that account exists
         let api = self.clone();
         api_spawn_blocking(move || {
-            let account = Account::new(api.context.clone(), address.0, None, None, None)?;
+            let account =
+                Account::new_with_internal_indexer(api.context.clone(), address.0, None, None)?;
```

Comment:
Why do we need `new_with_internal_indexer` after all? 
I think the Account should always respect the indexer version always. Otherwsie we can end up with returning wrong data because this line checks a passed in version against the the main db version https://github.com/aptos-labs/aptos-core/blob/872b134365f0dbec09981a5c002c4b89c0b84f86/api/src/accounts.rs#L207 

To make things easier I'd rather make the context always return a latest version from the indexer.

Reply:
Different purposes/usage require separating this to a new function. `new_with_internal_indexer` is only used for the APIs that internal indexer is used to get whatever the latest data without passing a ledger_version in argument. `new` is used when the ledger_version is passed for functions like https://github.com/aptos-labs/aptos-core/blob/get_latest_version_from_internal_indexer/api/src/accounts.rs#L69. getting the sequence number of the account,